src/chainstate/stacks/miner.rs

```diff
@@ -804,9 +1101,13 @@ impl StacksBlockBuilder {
         tx: &StacksTransaction,
         tx_len: u64,
         limit_behavior: &BlockLimitFunction,
-    ) -> Result<StacksTransactionReceipt, Error> {
+    ) -> TransactionResult {
         if self.bytes_so_far + tx_len >= MAX_EPOCH_SIZE.into() {
-            return Err(Error::BlockTooBigError);
+            return TransactionResult::skipped(
+                &tx,
+                "Adding this tx would exceed epoch data size".to_string(),
+                true,
+            );
```

Comment:
I don't know if this will actually impact the miner's control-flow, but these changes do alter the error returned by `try_mine_tx` in this case from `Error::BlockTooBigError` to `Error::StacksTransactionSkipped`. I think we may need to add an `Error` field to the `StacksTransactionSkipped` struct, which is just `Error::StacksTransactionSkipped` when constructed using `TransactionResult::skipped`, and then there's another variant like `TransactionResult::skipped_on_block_done()` that accepts an error like `Error::BlockTooBigError` as an argument.